spec.md

```diff
@@ -3149,6 +3167,227 @@ The CO MUST implement the specified error recovery behavior when it encounters t
+
+#### GetMetadataDelta RPC
+
+The plugin must implement this RPC if it provides the SnapshotMetadata service.
+
+```protobuf
+// The GetMetadataDeltaRequest message is used to solicit metadata on
+// the data ranges that have changed between two snapshots.
+message GetMetadataDeltaRequest {
+  // This is the identifier of the snapshot against which changes
+  // are to be computed.
+  // This field is REQUIRED.
+  string base_snapshot_id = 1;
+
+  // This is the identifier of a second snapshot in the same volume,
+  // created after the base snapshot.
+  // This field is REQUIRED.
+  string target_snapshot_id = 2;
+
+  // This indicates the zero based starting byte position in the volume
+  // snapshot from which the result should be computed. It is intended
+  // to be used to continue a previously interrupted call.
+  // The plugins may round down this offset to the nearest alignment
+  // boundary based on the BlockMetadataType used.
+  // This field is REQUIRED.
+  uint64 starting_offset = 3;
+
+  // This is an optional parameter, and if non-zero it specifies the
+  // maximum number of tuples to be returned in each
+  // GetMetadataDeltaResponse message returned by the RPC stream.
+  // The plugin will determine an appropriate value if 0, and is
+  // always free to send less than the requested value.
+  // This field is OPTIONAL.
+  uint32 max_results = 4;
+
+  // Secrets required by plugin to complete the request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 5 [(csi_secret) = true];
+}
+
+// GetMetadataDeltaResponse messages are returned in gRPC stream.
+// Cumulatively, they provide information on the data ranges that
+// have changed between the base and target snapshots specified
+// in the GetMetadataDeltaRequest message.
+message GetMetadataDeltaResponse {
+  // This specifies the style used in the BlockMetadata sequence.
+  // This value must be the same in all such messages returned by
+  // the stream.
+  // If block_metadata_type is FIXED_LENGTH, then the size_bytes field
+  // of each message in the block_metadata list MUST be constant.
+  // This field is REQUIRED.
+  BlockMetadataType block_metadata_type = 1;
+
+  // This returns the capacity of the underlying volume in bytes.
+  // This value must be the same in all such messages returned by
+  // the stream.
+  // This field is REQUIRED.
+  uint64 volume_capacity_bytes = 2;
+
+  // This is a list of data range tuples.
+  // If the value of max_results in the GetMetadataDeltaRequest message
+  // is non-zero, then the number of entries in this list MUST be less
+  // than or equal to that value.
+  // The byte_offset field of each message MUST be strictly increasing.
+  // The SP MUST NOT return a BlockMetadata message with a byte_offset
+  // less than than the value of starting_offset in the
```

Comment:
same conflict w/ earlier, optional rounding-down by plugin

Reply:
Changed to:

>  // The data range of the first BlockMetadata message in the list MUST
>  // include the starting_offset requested in the GetMetadataDeltaRequest
>  // message.

Reply:
@jdef Please see this comment below: https://github.com/container-storage-interface/spec/pull/551#issuecomment-1828864769